futaba/cogs/moderation/core.py

```diff
@@ -0,0 +1,243 @@
+#
+# cogs/moderation/core.py
+#
+# futaba - A Discord Mod bot for the Programming server
+# Copyright (c) 2017 Jake Richardson, Ammon Smith, jackylam5
+#
+# futaba is available free of charge under the terms of the MIT
+# License. You are free to redistribute and/or modify it under those
+# terms. It is distributed in the hopes that it will be useful, but
+# WITHOUT ANY WARRANTY. See the LICENSE file for more details.
+#
+
+# pylint: skip-file
+
+'''
+Collection of moderation commands such as Ban/Kick
+'''
+
+import asyncio
+import logging
+
+import discord
+from discord.ext import commands
+
+from futaba import permissions
+from futaba.enums import Reactions
+from futaba.utils import escape_backticks
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    'Moderation',
+]
+
+BAN_ACTION = discord.AuditLogAction.ban
+KICK_ACTION = discord.AuditLogAction.kick
+
+def user_disc(user):
+    ''' 
+    Return the user's username and disc
+    in the format username#disc
+    '''
+
+    return f'{user.name}#{user.discriminator}'
+
+class Moderation:
+    '''
+    Staff moderation commands
+    '''
+
+    __slots__ = (
+        'bot',
+        'journal',        
+    )
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.journal = bot.get_broadcaster('/moderation')
+
+    def __unload(self):
+        ''' Remove listeners '''
+
+        self.bot.remove_listener(self.member_ban, "on_member_ban")
+        self.bot.remove_listener(self.member_kick, "on_member_remove")
+
+    @commands.command()
+    @commands.guild_only()
+    @permissions.check_mod()
+    async def kick(self, ctx, member: discord.Member, *, reason: str):
+        '''
+        Kicks the user from the guild with a reason
+        If guild has moderation logging enabled, it is logged
+        '''
+
+        try:
+            embed = discord.Embed(description='Done! User Kicked')
+            embed.add_field(name='Reason:', value=reason)
+
+            mod = user_disc(ctx.author)
+            kicked = user_disc(member)
+            clean_reason = escape_backticks(reason)
+            content = f'{mod} kicked {member.mention} ({kicked}) with reason: `{clean_reason}`'
+            
+            await asyncio.gather(
+                ctx.guild.kick(member, reason=f'{reason} - {mod}'),
+                Reactions.SUCCESS.add(ctx.message),
+                ctx.send(embed=embed)
+            )
+
+            self.journal.send('member/kick', ctx.guild, content, icon='kick')
+        
+        except discord.errors.Forbidden:
+            
+            await asyncio.gather(
+                Reactions.DENY.add(ctx.message),
+                ctx.send("Can't do that user has higher role than me")
```

Comment:
Awkward phrasing: maybe "Can't kick somebody with a higher role than me", or simply just "I don't have permissions to do that"